src/bootstrap/doc.rs

```diff
@@ -384,10 +370,7 @@ impl Step for Standalone {
 
         // We open doc/index.html as the default if invoked as `x.py doc --open`
         // with no particular explicit doc requested (e.g. library/core).
-        if builder.paths.is_empty() || builder.was_invoked_explicitly::<Self>(Kind::Doc) {
-            let index = out.join("index.html");
-            open(builder, &index);
-        }
+        builder.maybe_open_in_browser::<Self>(out.join("index.html"));
```

Comment:
Hmm, I think the `paths.is_empty()` check is now missing, right? Can you test whether `x.py doc --open` with no arguments still opens the index page?

Instead I think you could keep the `invoked_explicitly` check for this function only and then call `open_in_browser` directly.

Reply:
Good catch, missed it.